qa/rpc-tests/sc_csw_eviction_from_mempool.py

```diff
@@ -0,0 +1,299 @@
+            {
+                "amount": sc_csw_amount,
+                "senderAddress": csw_mc_address,
+                "scId": scid2,
+                "epoch": 0,
+                "nullifier": nullifiers2[i],
+                "activeCertData": actCertData2,
+                "ceasingCumScTxCommTree": ceasingCumScTxCommTree2,
+                "scProof": csw_proofs2[i]
+            })
+
+        # recipient MC address, the same for all csw
+        taddr_2 = self.nodes[2].getnewaddress()
+        sc_csw_tx_outs = {taddr_2: Decimal(sc_csw_amount*10)}
+
+        rawtx = self.nodes[0].createrawtransaction([], sc_csw_tx_outs, sc_csws)
+        funded_tx = self.nodes[0].fundrawtransaction(rawtx)
+        sigRawtx = self.nodes[0].signrawtransaction(funded_tx['hex'], None, None, "NONE")
+        finalRawtx = self.nodes[0].sendrawtransaction(sigRawtx['hex'])
+        mark_logs("sent csw tx1 {}".format(finalRawtx), self.nodes, DEBUG_MODE)
+        self.sync_all()
+
+        mark_logs("Check tx is in mempool...", self.nodes, DEBUG_MODE)
+        assert_true(finalRawtx in self.nodes[2].getrawmempool())
+
+        # send another csw input for scid1 and verify it is rejected
+        sc_csw = [{
+            "amount": sc_csw_amount,
+            "senderAddress": csw_mc_address,
+            "scId": scid1,
+            "epoch": 0,
+            "nullifier": nullifiers1[-1],
+            "activeCertData": actCertData1,
+            "ceasingCumScTxCommTree": ceasingCumScTxCommTree1,
+            "scProof": csw_proofs1[-1]
+        }]
+
+        rawtx = self.nodes[0].createrawtransaction([], sc_csw_tx_outs, sc_csw)
+        funded_tx = self.nodes[0].fundrawtransaction(rawtx)
+        sigRawtx = self.nodes[0].signrawtransaction(funded_tx['hex'], None, None, "NONE")
+        mark_logs("Sending a further csw input to scid1, expecting failure...", self.nodes, DEBUG_MODE)
+        try:
+            finalRawtx2 = self.nodes[0].sendrawtransaction(sigRawtx['hex'])
+            assert(false)
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs("Send csw failed (expected) with reason {}".format(errorString), self.nodes, DEBUG_MODE)
+
+        # mine a block, in this way mempool is cleared and we can re-send the rejected tx
+        mark_logs("Node 1 generates 1 block", self.nodes, DEBUG_MODE)
+        blocks.extend(self.nodes[1].generate(1))
+        self.sync_all()
+
+        inv_hash = self.nodes[0].getbestblockhash()
+
+        mark_logs("Re-sending previously rejected tx with one more csw...", self.nodes, DEBUG_MODE)
+        try:
+            finalRawtx2 = self.nodes[0].sendrawtransaction(sigRawtx['hex'])
+            mark_logs("sent csw tx {}".format(finalRawtx), self.nodes, DEBUG_MODE)
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs("Send csw failed with reason {}".format(errorString), self.nodes, DEBUG_MODE)
+            assert(false)
```

Comment:
Typo, `false` is undefined.

```suggestion
            assert(False)
```

Reply:
Done